tests/firedrake/demos/test_demos_run.py

```diff
@@ -72,39 +70,32 @@ def test_no_missing_demos():
     assert not all_demo_locs, "Unrecognised demos listed"
 
 
-def _maybe_skip_demo(demo):
+def _maybe_skip_demo(demo, skip_dependency):
     # Add pytest skips for missing imports or packages
-    if "mumps" in demo.requirements and "mumps" not in get_external_packages():
+
+    if "mumps" in demo.requirements and skip_dependency("mumps"):
         pytest.skip("MUMPS not installed with PETSc")
 
-    if "hypre" in demo.requirements and "hypre" not in get_external_packages():
+    if "hypre" in demo.requirements and skip_dependency("hypre"):
```

Comment:
Could we turn this all into a loop:
```py
for dep in demo.requirements:
    if skip_dependency(dep):
        pytest.skip(reason=f"{dep} is not available")
```

Reply:
Probably should be. I think I didn't because the messages don't always line up with the dependency string, but that's not really a big problem.

Reply:
I think having a loop is better because then we can't accidentally add a requirement to a demo and then forget to add a check for it.

Reply:
Done